src/coprocessor/dag/batch/statistics.rs

```diff
@@ -18,20 +19,22 @@ pub struct BatchExecuteStatistics {
 }
 
 impl BatchExecuteStatistics {
+    /// Creates a new statistics instance.
+    ///
+    /// If execution summary does not need to be collected, it is safe to pass 0 to the `executors`
+    /// argument, which will avoid one allocation.
     pub fn new(executors: usize, ranges: usize) -> Self {
-        let mut summary_per_executor = Vec::new();
-        summary_per_executor.resize_with(executors, || None);
-
         Self {
-            summary_per_executor,
+            summary_per_executor: vec![ExecSummary::default(); executors],
```

Comment:
prefer to use `executors_len` and `ranges_len`

Reply:
Done, PTAL again, thanks!